src/renderer/modals/FullNode/steps/StepConnectDevice.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import React, { useCallback, useState, useEffect } from "react";
+import { reduce } from "rxjs/operators";
+import TrackPage from "~/renderer/analytics/TrackPage";
+import DeviceAction from "~/renderer/components/DeviceAction";
+import Button from "~/renderer/components/Button";
+import Box from "~/renderer/components/Box";
+import { shell } from "electron";
+import resolveUserDataDirectory from "~/helpers/resolveUserDataDirectory";
+import BigSpinner from "~/renderer/components/BigSpinner";
+import Text from "~/renderer/components/Text";
+import { Trans } from "react-i18next";
+import { createAction } from "@ledgerhq/live-common/lib/hw/actions/app";
+import { command } from "~/renderer/commands";
+import { getEnv } from "@ledgerhq/live-common/lib/env";
+import { getCryptoCurrencyById } from "@ledgerhq/live-common/lib/currencies";
+import { mockedEventEmitter } from "~/renderer/components/DebugMock";
+import type { FullNodeSteps, ConnectionStatus } from "~/renderer/modals/FullNode";
+import { CheckWrapper, connectionStatus } from "~/renderer/modals/FullNode";
+import IconCheck from "~/renderer/icons/Check";
+
+const connectAppExec = command("connectApp");
+const action = createAction(getEnv("MOCK") ? mockedEventEmitter : connectAppExec);
+
+const StepConnectDevice = ({
+  onStepChange,
+  setScannedDescriptors,
+  setError,
+}: {
+  onStepChange: FullNodeSteps => void,
+  setScannedDescriptors: any => void,
+  setError: Error => void,
+}) => {
+  const currency = getCryptoCurrencyById("bitcoin");
+  const [device, setDevice] = useState(null);
+  const [scanStatus, setScanStatus] = useState<ConnectionStatus>(connectionStatus.IDLE);
+
+  useEffect(() => {
+    if (device) {
+      const currency = getCryptoCurrencyById("bitcoin");
+      command("scanDescriptors")({
+        deviceId: device.deviceId,
+        currency,
+        limit: 10,
+      })
+        // $FlowFixMe I don't know what you want.
+        .pipe(reduce((acc, item) => acc.concat({ descriptor: item }), []))
+        .subscribe(descriptors => {
+          setScanStatus(connectionStatus.SUCCESS);
+          setScannedDescriptors(descriptors);
+        });
```

Comment:
⚠️ you must handle error case of the subscription which can happen for various reasons. Like sleep mode of device, quitting the app on the device, etc..